map/traffic_manager.cpp

```diff
@@ -93,3 +113,57 @@ void TrafficManager::CalculateSegmentsColoring(traffic::TrafficInfo const & traf
     output.emplace_back(sid, c.second);
   }
 }
+
+// static
+void TrafficManager::ThreadRoutine(TrafficManager * manager)
```

Comment:
You don't planning to pass null here, right? So, why not to pass TrafficManager by reference? Moreover, why do you need to pass manager at all???